packages/rpc-graphql/src/loaders/transaction.ts

```diff
@@ -9,85 +8,24 @@ import { GraphQLResolveInfo } from 'graphql';
 import type { Rpc } from '../context';
 import { TransactionQueryArgs } from '../schema/transaction';
 import { onlyPresentFieldRequested } from './common/resolve-info';
+import { transformLoadedTransaction } from './transformers/transaction';
 
-function normalizeArgs(args: TransactionQueryArgs) {
-    const { commitment, encoding, signature } = args;
+/* Normalizes RPC optional configs to use GraphQL API defaults */
+function normalizeArgs({ commitment = 'confirmed', encoding = 'jsonParsed', signature }: TransactionQueryArgs) {
```

Comment:
If you end up doing what I suggest [here](https://github.com/solana-labs/solana-web3.js/pull/1837#discussion_r1407188113) then the choice of encoding will only be `jsonParsed` if you have subselections that demand parsing, and otherwise it will take the fast path and do base64.

Reply:
Yeah, sounds good I'll roll the same approach here as well.